src/wesnoth.cpp

```diff
@@ -87,6 +87,7 @@
 
 #include <boost/iostreams/filtering_stream.hpp> // for filtering_stream
 #include <boost/program_options/errors.hpp>     // for error
+#include <boost/optional.hpp>
```

Comment:
Why boost::optional?

Reply:
Because all the other files I could find also used boost::optional. It's possible I was working on a branch before you replaced them somehow? Or I accidentally searched on the 1.14 branch, unlikely though that sounds... In any case, I can replace it.

Reply:
Yeah, please use `std::optional` unless you're using a reference type.

Reply:
Yeah, it was the use of `boost::optional` on references that confused me, as it turns out. Thus why I replaced them all with `utils::optional_reference` on master...  now the only place `boost::optional` is still used is in spirit_po.